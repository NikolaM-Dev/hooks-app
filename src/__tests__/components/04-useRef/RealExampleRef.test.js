import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import RealExampleRef from 'components/04-useRef/RealExampleRef';

describe('Test on <RealExampleRef /> component', () => {
  let wrapper = shallow(<RealExampleRef />);

  beforeEach(() => {
    wrapper = shallow(<RealExampleRef />);
  });

  it('should show <RealExampleRef /> correctly', () => {
    const multipleCustomHooksExists = wrapper
      .find('MultipleCustomHooks')
      .exists();

    expect(wrapper).toMatchSnapshot();
    expect(multipleCustomHooksExists).toBeFalsy();
  });

  it('should show the <MultipleCustomHooks /> component', () => {
    const button = wrapper.find('button');
    button.simulate('click');
    const isMultipleCustomHooks = wrapper.find('MultipleCustomHooks').exists();

    expect(isMultipleCustomHooks).toBeTruthy();
  });
});
