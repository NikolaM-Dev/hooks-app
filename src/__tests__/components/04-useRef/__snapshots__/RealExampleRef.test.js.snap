// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <RealExampleRef /> component should show <RealExampleRef /> correctly 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
